packages/flutter_adaptive_scaffold/lib/src/breakpoints.dart

```diff
@@ -129,3 +131,49 @@ abstract class Breakpoint {
   /// the screen such as MediaQuery.sizeOf(context).width.
   bool isActive(BuildContext context);
 }
+
+/// An extension on [BuildContext] that returns the active [Breakpoint] based on
+/// an available [SlotLayout] ancestor or the default breakpoints.
+extension BreakpointExtension on BuildContext {
+  /// Returns the currently active [Breakpoint].
+  Breakpoint get activeBreakpoint {
+    final SlotLayout? slotLayout = findAncestorWidgetOfExactType<SlotLayout>();
+    if (slotLayout != null) {
+      for (final MapEntry<Breakpoint, SlotLayoutConfig?> config
+          in slotLayout.config.entries) {
+        if (config.key.isActive(this)) {
+          return config.key;
+        }
+      }
+    }
+
+    final TargetPlatform platform = Theme.of(this).platform;
+
+    if (Breakpoint.desktop.contains(platform)) {
+      if (Breakpoints.extraLarge.isActive(this)) {
+        return Breakpoints.extraLargeDesktop;
+      } else if (Breakpoints.large.isActive(this)) {
+        return Breakpoints.largeDesktop;
+      } else if (Breakpoints.mediumLarge.isActive(this)) {
+        return Breakpoints.mediumLargeDesktop;
+      } else if (Breakpoints.medium.isActive(this)) {
+        return Breakpoints.mediumDesktop;
+      } else if (Breakpoints.small.isActive(this)) {
+        return Breakpoints.smallDesktop;
+      }
+    } else if (Breakpoint.mobile.contains(platform)) {
+      if (Breakpoints.extraLarge.isActive(this)) {
+        return Breakpoints.extraLargeMobile;
+      } else if (Breakpoints.large.isActive(this)) {
+        return Breakpoints.largeMobile;
+      } else if (Breakpoints.mediumLarge.isActive(this)) {
+        return Breakpoints.mediumLargeMobile;
+      } else if (Breakpoints.medium.isActive(this)) {
+        return Breakpoints.mediumMobile;
+      } else if (Breakpoints.small.isActive(this)) {
+        return Breakpoints.smallMobile;
+      }
+    }
+    return Breakpoints.standard;
```

Comment:
So, if I set my own custom breakpoint, `activeBreakpoint` won't return it, right? And in fact, it will return one that is different from the actual value.  It think it would be better to return null here, and rename it to something that indicates that it's one of the standard ones (e.g. activeStandardBreakpoint) or something. And while this is still useful, it's more complicated to understand what it does.

Ideally, you'd find out the actual active breakpoint somehow and return that rather than guessing that it's `Breakpoints.standard`.  That would be easy to use and understand.

Reply:
Well you can only set actual breakpoints when using a SlotLayout somewhere. So this looks that up, but if you don't use a SlotLayout still returns the currently active config.